packages/hardhat-core/src/internal/hardhat-network/provider/node.ts

```diff
@@ -162,6 +175,19 @@ export class HardhatNode extends EventEmitter {
       initialBlockTimeOffset = new BN(
         getDifferenceInSeconds(new Date(forkBlockTimestamp), new Date())
       );
+
+      if (hardforkGte(hardfork, HardforkName.LONDON)) {
```

Comment:
I think this block merits a comment. Something like:

> In some scenarios we set the `baseFeePerGas` of the first block that will be mined after the fork

(Which scenarios are these is clear enough after reading the code)

Reply:
Done